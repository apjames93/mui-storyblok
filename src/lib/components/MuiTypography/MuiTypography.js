import React from 'react';
import PropTypes from 'prop-types';
import TypographyText from '../TypographyText/TypographyText';
import MuiIcon from '../MuiIcon/MuiIcon';
import MuiIconButtonRedirect from '../MuiIconButtonRedirect/MuiIconButtonRedirect';
import MuiIconButtonHref from '../MuiIconButtonHref/MuiIconButtonHref';
import MuiIconButtonDownload from '../MuiIconButtonDownload/MuiIconButtonDownload';
import MuiButtonRedirect from '../MuiButtonRedirect/MuiButtonRedirect';
import MuiButtonHref from '../MuiButtonHref/MuiButtonHref';
import MuiButtonDownload from '../MuiButtonDownload/MuiButtonDownload';
import MuiLink from '../MuiLink/MuiLink';
import MuiButtonDialog from '../MuiButtonDialog/MuiButtonDialog';
import MuiIconButtonDialog from '../MuiIconButtonDialog/MuiIconButtonDialog';
import MuiTooltip from '../MuiTooltip/MuiTooltip';

/**
 * MuiTypography is used in storyblok redirect to react routes
 */

export const MuiTypography = (props) => {
  const components = {
    MuiLink,
    MuiIcon,
    MuiIconButtonRedirect,
    MuiIconButtonHref,
    MuiIconButtonDownload,
    MuiIconButtonDialog,
    MuiButtonRedirect,
    MuiButtonHref,
    MuiButtonDownload,
    MuiButtonDialog,
    MuiTooltip,
  };

  return (
    <TypographyText
      {...props}
      components={components}
    />
  );
};

export default MuiTypography;

MuiTypography.propTypes = {

  content: PropTypes.arrayOf(PropTypes.shape({
    component: PropTypes.string.isRequired,
  })).isRequired,
};

MuiTypography.defaultProps = {

};
