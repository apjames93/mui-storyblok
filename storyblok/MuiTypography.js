module.exports = {
  "name": "MuiTypography",
  "display_name": null,
  "updated_at": "2020-07-14T17:25:27.261Z",
  "schema": {
    "align": {
      "type": "option",
      "use_uuid": true,
      "options": [
        {
          "value": "inherit",
          "name": "inherit"
        },
        {
          "value": "left",
          "name": "left"
        },
        {
          "value": "center",
          "name": "center"
        },
        {
          "value": "right",
          "name": "right"
        },
        {
          "value": "justify",
          "name": "justify"
        }
      ],
      "default_value": "inherit",
      "required": true
    },
    "color": {
      "type": "option",
      "use_uuid": true,
      "options": [
        {
          "value": "initial",
          "name": "initial"
        },
        {
          "value": "inherit",
          "name": "inherit"
        },
        {
          "value": "primary",
          "name": "primary"
        },
        {
          "value": "secondary",
          "name": "secondary"
        },
        {
          "value": "textPrimary",
          "name": "textPrimary"
        },
        {
          "value": "textSecondary",
          "name": "textSecondary"
        },
        {
          "value": "error",
          "name": "error"
        }
      ],
      "default_value": "initial",
      "required": true
    },
    "rootClass": {
      "type": "options",
      "options": [
        {
          "value": "margin-right: 15px",
          "name": "margin-right"
        },
        {
          "value": "margin-bottom: 15px",
          "name": "margin-bottom"
        },
        {
          "value": "opacity: .7",
          "name": "opacity"
        },
        {
          "value": "background-color: #FFF",
          "name": "bc white"
        }
      ],
      "folder_slug": "",
      "description": "",
      "display_name": "",
      "translatable": false,
      "required": false,
      "external_datasource": ""
    },
    "width": {
      "type": "text",
      "required": true,
      "default_value": "100%"
    },
    "height": {
      "type": "text",
      "required": true,
      "default_value": "100%"
    },
    "content": {
      "type": "bloks",
      "maximum": "",
      "restrict_components": true,
      "component_whitelist": [
        "MuiLink",
        "MuiIcon",
        "MuiIconButtonRedirect",
        "MuiIconButtonHref",
        "MuiIconButtonDownload",
        "MuiIconButtonDialog",
        "MuiButtonRedirect",
        "MuiButtonDownload",
        "MuiButtonDialog",
        "MuiButtonHref",
        "MuiText",
        "MyCoolComp",
        "GoogleConversions"
      ],
      "required": true
    }
  },
  "image": null,
  "preview_field": null,
  "is_root": false,
  "preview_tmpl": null,
  "is_nestable": true,
  "all_presets": [],
  "preset_id": null,
  "real_name": "MuiTypography",
  "component_group_uuid": null
}