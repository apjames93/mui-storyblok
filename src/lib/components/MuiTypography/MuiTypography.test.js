import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import MuiTypography from './MuiTypography';

function setup() {
  const props = {
    _editable: '<!--#storyblok#{"name": "MuiTypography", "space": "48408", "uid": "947be9f0-47c3-4315-a95a-550f0c560eb5", "id": "307934"}-->',
    content: [{
      _editable: '<!--#storyblok#{"name": "MuiText", "space": "48408", "uid": "947be9f0-47c3-4315-a95a-550f0c560eb5", "id": "307934"}-->',
      component: 'MuiText',
      text: 'this is text',
    }, {
      _editable: '<!--#storyblok#{"name": "MuiLink", "space": "48408", "uid": "947be9f0-47c3-4315-a95a-550f0c560eb5", "id": "307934"}-->',
      component: 'MuiLink',
      text: 'btn',
      href: 'google.com',
    }, {
      _editable: '<!--#storyblok#{"name": "MuiIconButtonRedirect", "space": "48408", "uid": "947be9f0-47c3-4315-a95a-550f0c560eb5", "id": "307934"}-->',
      component: 'MuiIconButtonRedirect',
      redirectRoute: 'page-welcom',
      iconButton: [{
        _editable: '<!--#storyblok#{"name": "MuiIconButton", "space": "48408", "uid": "947be9f0-47c3-4315-a95a-550f0c560eb5", "id": "307934"}-->',
        component: 'MuiIconButton',
        icon: [{
          _editable: '<!--#storyblok#{"name": "MuiIcon", "space": "48408", "uid": "947be9f0-47c3-4315-a95a-550f0c560eb5", "id": "307934"}-->',
          component: 'MuiIcon',
          iconName: 'star',
        }],
      }],
    }, {
      _editable: '<!--#storyblok#{"name": "MuiIcon", "space": "48408", "uid": "947be9f0-47c3-4315-a95a-550f0c560eb5", "id": "307934"}-->',
      component: 'MuiIcon',
      iconName: 'android',
    }],
  };
  const comp = shallow(<MuiTypography {...props} />);
  return { comp, props };
}

describe('<MuiTypography />', () => {
  it('renders MuiTypography', () => {
    const { comp } = setup();
    expect(comp).toBeDefined();
  });

  test('snapshot', () => {
    const { props } = setup();

    const tree = renderer.create(<MuiTypography {...props} />);
    expect(tree).toMatchSnapshot();
  });
});
