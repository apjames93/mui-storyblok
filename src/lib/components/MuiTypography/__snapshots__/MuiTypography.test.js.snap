// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MuiTypography /> snapshot 1`] = `
<div
  className="MuiTypography-root makeStyles-root-1 MuiTypography-body1"
>
  <span
    className="MuiTypography-root makeStyles-root-2 MuiTypography-body1"
  >
     undefined 
  </span>
  <a
    className="MuiTypography-root MuiLink-root MuiLink-underlineHover makeStyles-root-3 MuiTypography-colorPrimary"
    href="#"
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
  >
    btn
  </a>
  <button
    className="MuiButtonBase-root MuiIconButton-root makeStyles-root-4"
    data-testid="muiIconButton"
    disabled={false}
    onBlur={[Function]}
    onClick={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={0}
    type="button"
  >
    <span
      className="MuiIconButton-label"
    >
      <span
        aria-hidden={true}
        className="material-icons MuiIcon-root makeStyles-root-5 MuiIcon-colorSecondary"
      >
        star
      </span>
    </span>
  </button>
  <span
    aria-hidden={true}
    className="material-icons MuiIcon-root makeStyles-root-6 MuiIcon-colorSecondary"
  >
    android
  </span>
</div>
`;
